sdk/loadtestservice/azure-developer-loadtesting/tests/conftest.py

```diff
@@ -0,0 +1,62 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import os
+import platform
+import pytest
+import sys
+
+from dotenv import load_dotenv
+
+from devtools_testutils import test_proxy, add_general_regex_sanitizer, add_body_key_sanitizer, add_header_regex_sanitizer
+
+# Ignore async tests for Python < 3.5
+collect_ignore_glob = []
+if sys.version_info < (3, 5) or platform.python_implementation() == "PyPy":
+    collect_ignore_glob.append("*_async.py")
+
+load_dotenv()
+
+@pytest.fixture(scope="session", autouse=True)
+def add_sanitizers(test_proxy):
```

Comment:
I think you're having to add all of these sanitizers because you're not properly handling the environment variables. They should be done through the powershell preparer. [Here's](https://github.com/Azure/azure-sdk-for-python/blob/fe6841289a45be0e89404b04332b8f7b78921c16/sdk/deviceupdate/azure-iot-deviceupdate/tests/testcase.py#L23) an example, here's the testing doc https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/tests.md#deliver-environment-variables-to-tests

Reply:
Like @iscai-msft said, providing environment variables to the `EnvironmentVariableLoader`/`PowerShellPreparer` will automatically register sanitizers for the provided values. The `tests.md` documentation doesn't actually seem to mention this explicitly, but the [test proxy-specific documentation](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/test_proxy_migration_guide.md#fetch-environment-variables) does.

So, if you want to add your sub ID, client ID, etc. to the `LoadtestingPowerShellPreparer`, that should make these sanitizer requests unnecessary; but I personally don't have an issue with your current approach either 🙂